xbmc/video/VideoDimensions.cpp

```diff
@@ -0,0 +1,96 @@
+/*
+ *      Copyright (C) 2005-2013 Team XBMC
+ *      http://xbmc.org
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "VideoDimensions.h"
+#include <algorithm>
+
+std::map<std::string, Boundary> CVideoDimensions::boundaries =
+{
+  { "480", { { 0, 720 }, { 0, 480 } } },
+  { "576", { { 0, 720 }, { 480, 576 } } },
+  { "540", { { 720, 960 }, { 540, 544 } } },
+  { "720", { { 960, 1280 }, { 540, 720 } } },
+  // Full-SBS 3D can be 3840x1080
+  { "1080", { { 1280, 3840 }, { 540, 1080 } } },
+  // Full-SBS 4K 3D could be 8192x2160
+  { "4K", { { 2048, 8192 }, { 1080, 2160 } } }
+};
+
+std::map<std::string, std::string> CVideoDimensions::fidelities =
+{
+  { "480", "SD" },
+  { "576", "SD" },
+  { "540", "SD" },
+  { "720", "HD" },
+  { "1080", "HD" },
+  { "4K", "HD" }
+};
+
+CVideoDimensions::CVideoDimensions(int width, int height) :
+m_width(width),
+m_height(height)
+{
+}
+
+std::string CVideoDimensions::GetDescriptor() const
+{
+  auto it = std::find_if(
+    boundaries.cbegin(),
+    boundaries.cend(),
+    [=](const std::pair<std::string, Boundary> boundary)
+  {
+    auto width = boundary.second.first;
+    auto height = boundary.second.second;
+
+    return m_width > width.first && m_width <= width.second &&
+      m_height > height.first && m_height <= height.second;
+  });
+
+  if (it != boundaries.cend())
+    return it->first;
+
+  return "";
+}
+
+std::string CVideoDimensions::GetFidelity() const
+{
+  std::string descriptor = GetDescriptor();
+
+  for (const auto &fidelity : fidelities)
+    if (fidelity.first == descriptor)
+      return fidelity.second;
+
+  return "";
+}
+
+std::vector<std::string> CVideoDimensions::GetDescriptors()
+{
+  std::vector<std::string> descriptors;
+
+  for (const auto &boundary : CVideoDimensions::boundaries)
+    descriptors.push_back(boundary.first);
+
+  return descriptors;
+}
+
+Boundary CVideoDimensions::GetBoundary(const std::string &descriptor)
+{
+  return CVideoDimensions::boundaries[descriptor];
```

Comment:
No check if the `descriptor` exists in the map?
And you should be able to drop the `CVideoDimensions::`.